-1

Tested at: 7a8aadb648f0763dbb23b4c25e84a43d51a00ce1

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-22274/26192/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/FedConnections.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/AnalysisDescriptions.cc 
>> Compiling edm plugin /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/plugins/module.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc: In member function 'void SiStripConfigDb::usingXmlFiles()':
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc:615:13: error: 'std::vector<std::__cxx11::basic_string<char> >::iterator {aka class __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >}' has no member named 'empty'
  if ( *iter.empty() ) {
             ^~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_1_DEVEL_X_2018-02-19-2300/src/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc:645:13: error: 'std::vector<std::__cxx11::basic_string<char> >::iterator {aka class __gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >}' has no member named 'empty'
  if ( *iter.empty() ) {
             ^~~~~
</pre>

--- a/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc
+++ b/OnlineDB/SiStripConfigDb/src/SiStripConfigDb.cc
@@ -612,7 +612,7 @@ void SiStripConfigDb::usingXmlFiles() {
     } else {
       std::vector<std::string>::iterator iter = ip->second.inputFecXml().begin();
       for ( ; iter != ip->second.inputFecXml().end(); iter++ ) {
-	if ( *iter.empty() ) {
+	if ( (*iter).empty() ) {
 	  edm::LogWarning(mlConfigDb_)
 	    << "[SiStripConfigDb::" << __func__ << "]"
 	    << " NULL path to input 'fec.xml' file!";
@@ -642,7 +642,7 @@ void SiStripConfigDb::usingXmlFiles() {
     } else {
       std::vector<std::string>::iterator iter = ip->second.inputFedXml().begin();
       for ( ; iter != ip->second.inputFedXml().end(); iter++ ) {
-	if ( *iter.empty() ) {
+	if ( (*iter).empty() ) {
 	  edm::LogWarning(mlConfigDb_) 
 	    << "[SiStripConfigDb::" << __func__ << "]"
 	    << " NULL path to input 'fed.xml' file!";

